src/Bicep.Core.IntegrationTests/NestedRuntimeMemberAccessTests.cs

```diff
@@ -37,15 +37,33 @@ public void Nested_runtime_member_access_is_not_supported()
   name: appPlan.properties.provisioningState
 }
 
-var foo = virtualMachine::windowsVMDsc.properties.autoUpgradeMinorVersion
+resource vm2 'Microsoft.Compute/virtualMachines@2020-12-01' existing = {
+  name: appPlan.properties.workerTierName // Create a runtime reference.
+}
+
+resource windowsVMDsc2 'Microsoft.Compute/virtualMachines/extensions@2020-12-01' existing = {
+  name: 'vmDsc2'
+  parent: vm2
+}
+
+resource diag 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' existing = {
+  name: 'diag'
+  scope: vm2
+}
+
+var foo = vm::windowsVMDsc.properties.autoUpgradeMinorVersion
 var bar = storage.properties.accessTier
 var baz = storage.listKeys().keys
+var qux = windowsVMDsc2.properties
+var quux = diag.properties
```

Comment:
Do these variables need to be declared to trigger any of the errors below? They look like they would all need to be inlined but should be legal.

Reply:
Yes, they trigger the last two errors. Accessing properties of `windowsVMDsc2` and `diag` will generate nested references, even when they are inlined, because their `parent` and `scope` resource (`vm2`) contains a runtime reference, so their IDs will also contain the runtime reference.

Reply:
I wrongly assumed in asking my question above that inlined variables that go unused wouldn't generate any diagnostics since they wouldn't appear in the compiled template.